x-pack/plugins/canvas/public/functions/pie.ts

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, keyBy, map, groupBy } from 'lodash';
+import { PaletteOutput } from 'src/plugins/charts/public';
+import { InitializeArguments } from './';
+// @ts-expect-error untyped local
+import { getLegendConfig } from '../../common/lib/get_legend_config';
+import { getFunctionHelp } from '../../i18n';
+import {
+  Legend,
+  PointSeries,
+  Render,
+  SeriesStyle,
+  Style,
+  ExpressionFunctionDefinition,
+} from '../../types';
+
+interface PieSeriesOptions {
+  show: boolean;
+  innerRadius: number;
+  stroke: {
+    width: number;
+  };
+  label: {
+    show: boolean;
+    radius: number;
+  };
+  tilt: number;
+  radius: number | 'auto';
+}
+
+interface PieOptions {
+  canvas: boolean;
+  colors: string[];
+  legend: {
+    show: boolean;
+    backgroundOpacity?: number;
+    labelBoxBorderColor?: string;
+    position?: Legend;
+  };
+  grid: {
+    show: boolean;
+  };
+  series: {
+    pie: PieSeriesOptions;
+  };
+}
+
+interface PieData {
+  label: string;
+  data: number[];
+  color?: string;
+}
+
+export interface Pie {
+  font: Style;
+  data: PieData[];
+  options: PieOptions;
+}
+
+interface Arguments {
+  palette: PaletteOutput;
+  seriesStyle: SeriesStyle[];
+  radius: number | 'auto';
+  hole: number;
+  labels: boolean;
+  labelRadius: number;
+  font: Style;
+  legend: Legend | false;
+  tilt: number;
+}
+
+export function pieFunctionFactory(
+  initialize: InitializeArguments
```

Comment:
I think it makes more sense to pass the Palette Service to this function, rather than `InitializeArguments`.  The contract is much more explicit.